src/libNode/Node.cpp

```diff
@@ -473,12 +473,13 @@ bool Node::StartRetrieveHistory(const SyncType syncType,
       (m_mediator.m_txBlockChain.GetBlockCount()) % NUM_FINAL_BLOCK_PER_POW;
 
   /// Save coin base for micro block, from last DS epoch to current TX epoch
-  if (!LOOKUP_NODE_MODE) {
+  if (bDS) {
     std::list<MicroBlockSharedPtr> microBlocks;
     if (BlockStorage::GetBlockStorage().GetRangeMicroBlocks(
             m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetEpochNum(),
-            m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetBlockNum(),
-            0, m_mediator.m_ds->m_shards.size(), microBlocks)) {
+            m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetBlockNum() +
+                1,
+            0, m_mediator.m_ds->m_shards.size() - 1, microBlocks)) {
```

Comment:
Previous implementation is wrong because for DS nodes, we only save coin base with shardId = [0, shardSize -1], no shardId = shardSize. Thus the range should be subtracted by 1.

Reply:
but we store all the microblock. so even we use only 0 - shardSize-1 to calculate the coinbase, the selection should still be done in coinbase level rather than the fetching of microblock